### PR TITLE
Refactor/promisify strategies

### LINKED ISSUE #25: Convert storage strategies to promise apis to allow async await support
We list node 8.1+ as our requirement, so we're free to depend on promises and async await.

We should make use of this for our storage strategies.

### PR DESCRIPTION
**Pull request by TheMallen**
Closes #25

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,11 +34,11 @@
     "connect-redis": "^3.3.0",
     "express": "^4.16.2",
     "express-session": "^1.15.3",
+    "handy-redis": "^1.3.0",
```
Comment (AWaselnuk) [inline package.json]: What does this thing give us?

Comment (IvanLinos) [reply package.json]: why dont just promisify redis get/sets instead depending on a 3rd party module which is could not be maintained in the future?

Example:
```
const client = redis.createClient();
const util = require('util');
const asyncGet = util.promisify(client.get).bind(client);
```

**Comment by jamiemtdwyer** [comment]
I 🎩'd and everything works as expected, but I am hesitant about including `handy-redis` as a dependency

File: package.json (hunk 1 above)
Comment (TheMallen) [reply package.json]: @AWaselnuk  

`handy-redis` is basically just doing the same thing Ivan is suggesting internally.

I don't have super strong feelings about it so I'll probably switch it to a manually promisified implementation.

**Comment by jamiemtdwyer** [comment]
LGTM! I 🎩'd the auth flow as well as the login flow, then logged in again in an incognito window and everything works as expected.

**Pull request merged**